### DISCUSSION TITLE
Select relationship data from table not from picker

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#6253] [Help]
I have two bbdb tables to store `information` (e.g. lead, order, invoice) and `systems` which process/store this information (e.g. ERP, CRM, Mail, Fileserver). On the `information` create or update screen, user can select one or more `systems` based on a many-to-many relation between the tables using a relationship picker.

While using the relationship picker does work and is VERY convenient (easy to implement), the list of systems has become to long (> 50 entries) for easy scrolling and user start to "miss" relevant entries or complain about poor readability (which includes the `information`read screen).

What would be the best way to display a searchable/filterable table of related data instead of a relationship picker?

**Comment by melohagan** [comment]
Hi @gerhardmpl 

You could try this:

1. Add a table that displays the relational data: 
    https://docs.budibase.com/docs/fetching-data#displaying-relational-data
2. Add a search box for the system name:
    https://docs.budibase.com/docs/adding-a-search-field
3. *Allow row selection* in the table. This will allow you to use the `{{ <table-name>.Selected rows }}` binding when you are saving your form. 

It should be noted that you will only be able to select as many relationships as are available based on pagination.
So if you need to select many relationships, something like 10+, then it may be wise to use <a href="https://docs.budibase.com/docs/app-state">App State</a> to store a "basket" of relationships, with add/remove buttons per row: 
https://docs.budibase.com/docs/button#adding-buttons-to-table-rows 

Then you would use the App State binding instead of the Selected rows binding.
